File: sample-videochat-webrtc/sample-videochat-webrtc/Components/Users/Views/SelectedUsersCountAlert/SelectedUsersCountAlert.m
//
//  SelectedUsersCountAlert.m
//  sample-videochat-webrtc
//
//  Created by Injoit on 04.08.2021.
//  Copyright © 2021 QuickBlox Team. All rights reserved.
//

#import "SelectedUsersCountAlert.h"
#import "PaddingLabel.h"
#import "UIView+Videochat.h"

@interface SelectedUsersCountAlert ()
//MARK: - IBOutlets
@property (weak, nonatomic) IBOutlet PaddingLabel *alertLabel;
@property (weak, nonatomic) IBOutlet UIView *alertView;

@end

@implementation SelectedUsersCountAlert
//MARK: - Life Cycle
- (void)viewDidLoad {
    [super viewDidLoad];
    
    [self setupViews];
}

//MARK: - Actions
- (IBAction)tapCancelButton:(UIButton *)sender {
    [self dismissViewControllerAnimated:NO completion:nil];
}

//MARK: - Private Methods
- (void)setupViews {
    [self.alertLabel setupTextPaddingInsets: UIEdgeInsetsMake(0.0f, 11.0f, 0.0f, 0.0f)];
    [self.alertLabel setRoundViewWithCornerRadius:3.0f];
    [self.alertView addShadow:[UIColor colorWithRed:0.2f green:0.2f blue:0.2f alpha:1.0f]];
    
    CGFloat topBarHeight = self.view.window.windowScene.statusBarManager.statusBarFrame.size.height + self.navigationController.navigationBar.frame.size.height;
    self.alertView.translatesAutoresizingMaskIntoConstraints = NO;
    [self.alertView.leftAnchor constraintEqualToAnchor:self.view.leftAnchor constant:12.0f].active = YES;
    [self.alertView.topAnchor constraintEqualToAnchor:self.view.safeAreaLayoutGuide.topAnchor constant:112.0f + topBarHeight].active = YES;
    [self.alertView.rightAnchor constraintEqualToAnchor:self.view.rightAnchor constant:-12.0f].active = YES;
    [self.alertView.heightAnchor constraintEqualToConstant:44.0f].active = YES;
}

@end

File: sample-videochat-webrtc-swift/sample-videochat-webrtc-swift/Components/Users/SelectedUsersCountAlert/SelectedUsersCountAlert.swift
//
//  MaxCountAlertViewController.swift
//  sample-videochat-webrtc-swift
//
//  Created by Injoit on 29.07.2021.
//  Copyright © 2021 QuickBlox. All rights reserved.
//

import UIKit

struct SelectedUsersCountAlertConstant {
    static let defaultNavigationBarHeight: CGFloat = 44.0
}

class SelectedUsersCountAlert: UIViewController {
    //MARK: - IBOutlets
    @IBOutlet weak var alertLabel: PaddingLabel! {
        didSet {
            alertLabel.textPaddingInsets = UIEdgeInsets(top: 0, left: 11, bottom: 0, right: 0)
            alertLabel.setRoundView(cornerRadius: 3.0)
        }
    }
    @IBOutlet weak var alertView: UIView! {
        didSet {
            alertView.addShadowToView(color: #colorLiteral(red: 0.1960526407, green: 0.1960932612, blue: 0.1960500479, alpha: 1))
        }
    }

    //MARK: - Life Cycle
    override func viewDidLoad() {
        super.viewDidLoad()
        
        setupViews()
    }

    //MARK: - Actions
    @IBAction func tapCancelButton(_ sender: Any) {
        dismiss(animated: false)
    }
    
    //MARK: - Private Methods
    private func setupViews() {
        let topBarHeight = navigationController?.navigationBar.frame.height ?? SelectedUsersCountAlertConstant.defaultNavigationBarHeight
        alertView.translatesAutoresizingMaskIntoConstraints = false
        alertView.leftAnchor.constraint(equalTo: view.leftAnchor, constant: 12.0).isActive = true
        alertView.topAnchor.constraint(equalTo: view.safeAreaLayoutGuide.topAnchor, constant: 112.0 + topBarHeight).isActive = true
        alertView.leftAnchor.constraint(equalTo: view.leftAnchor, constant: 12.0).isActive = true
        alertView.heightAnchor.constraint(equalToConstant: 44.0).isActive = true  
    }
}
